jekyll/_cci2/manage-contexts-with-config-policies.adoc

```diff
@@ -0,0 +1,529 @@
+
+You can also push to a development branch, in which case you will get a diff of your policy bundle when you push your changes, rather than your new policy being pushed to your CircleCI organization. This is useful when developing your policies.
+--
+
+NOTE: If you would like to write tests for your policy, check out the xref:test-config-policies#[Test config policies] guide.
+
+[#conclusion-2]
+=== Conclusion
+Once you have pushed your new `allow_contexts.rego` policy, if an attempt to trigger a pipeline is made, in which the specified project has access to contexts in the block-list configured in your policy, the pipeline will fail to trigger. Developers will be notified on the dashboard as shown below.
+
+image::config-policies/context-fail-2.png[Dashboard page showing fail]
+
+[#define-the-contexts-reserved-by-a-project]
+== Define the contexts reserved by a project
+
+You may want to reserve contexts for use by a defined list of projects, blocking the use of those contexts by any project not in the allow-list. One possible use case for this would be locking contexts related to OIDC access to only those applications (projects) that need it. Any app that does not need this OIDC access will not be able to access those contexts. Developers will receive a hard fail, and pipelines will fail to trigger.
+
+This helper function accepts project IDs and context names. It prevents the usage of any specified context for projects that are not in the allow-list.
+
+[#create-your-policy-3]
+=== 1. Create your policy
+
+. If you have not already done so, create an empty directory to store your policies. For example:
++
+[source,shell]
+----
+mkdir ./config-policies
+----
+
+. Create a new file in this directory for your new policy. Name the file `reserve_contexts.rego`.
+
+. Copy the following snippet into the `reserve_contexts.rego` file you just made:
++
+[source,rego]
+----
+# All policies start with the org package definition
+package org
+
+# import CircleCI specific helper functions
+import data.circleci.config
+
+# Declare a policy name
+policy_name["reserved_contexts"]
+
+# Declare a rule
+rule_reserve_contexts = config.contexts_reserved_by_project_ids{
+    ["<project-ID-1>","<project-ID-1>"],
+    ["<context-1>","<context-2>"]
+}
+
+# Enable the rule and choose the hard_fail enforcement level
+enable_hard["rule_reserve_contexts"]
+----
++
+NOTE: Notice the `contexts_reserved_by_project_ids` helper is accessed using the `config` keyword.
++
+In the following steps you will replace `<project-ID-1>` and `<context-1>` (and `<project-ID-2>` and `<context-2>` if you choose to use more than one) with _your_ projects and contexts. The `reserve_contexts.rego` policy, once uploaded, will restrict your specified project so that it will not have access to your specified context(s).
```

Comment:
```suggestion
In the following steps you will replace `<project-ID-1>` and `<context-1>` (and `<project-ID-2>` and `<context-2>` if you choose to use more than one) with _your_ projects and contexts. The `reserve_contexts.rego` policy, once uploaded, will restrict your specified context(s) so that they can only be used by the project you have allowed.
```